server/src/main/java/org/candlepin/model/AbstractHibernateObject.java

```diff
@@ -33,12 +33,17 @@
 
 /**
  * Abstract class for hibernate entities
+ *
+ * @param <T>
+ *  Entity type extending this class; should be the name of the subclass
  */
 @MappedSuperclass
 @XmlType(name = "CandlepinObject")
 @JsonIgnoreProperties(ignoreUnknown = true)
 @JsonFilter("DefaultFilter")
-public abstract class AbstractHibernateObject implements Persisted, Serializable {
+public abstract class AbstractHibernateObject<T extends AbstractHibernateObject>
```

Comment:
The seemingly useless generic here are present to help work around the covarience in the parameters thing I was running into. They basically act as a compile-time marker to ensure that we don't accidentally overload our methods, and ensure we're still passing the right thing in.